Carbon.IIS/Functions/Remove-CIisCollectionItem.ps1

```diff
@@ -0,0 +1,150 @@
+        [String] $CollectionName,
+
+        # The value to be removed.
+        [Parameter(Mandatory, ValueFromPipeline)]
+        [String[]] $Value,
+
+        # ***INTERNAL***. Do not use.
+        [switch] $SkipCommit
+    )
+
+    begin
+    {
+        Set-StrictMode -Version 'Latest'
+        Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+        $getArgs = @{}
+        if ($CollectionName)
+        {
+            $getArgs['Name'] = $CollectionName
+        }
+
+        $displayPath = ''
+        if ($ConfigurationElement)
+        {
+            $getArgs['ConfigurationElement'] = $ConfigurationElement
+            $displayPath = $ConfigurationElement.ElementTagName
+        }
+        else
+        {
+            $getArgs['SectionPath'] = $SectionPath
+            if ($LocationPath)
+            {
+                $getArgs['LocationPath'] = $LocationPath
+            }
+            $displayPath =
+                Get-CIisDisplayPath -SectionPath $SectionPath -LocationPath $LocationPath -SubSectionPath $CollectionName
+        }
+
+        $stopProcessing = $false
+
+        $collection = Get-CIisCollection @getArgs
+        if (-not $collection)
+        {
+            $stopProcessing = $true
+            return
+        }
+
+        $keyAttrName = Get-CIisCollectionKeyName -Collection $collection
+
+        if (-not $keyAttrName)
+        {
+            $stopProcessing = $true
+            $msg = "Failed to remove items from IIS configuration collection ${displayPath} because that collection " +
+                   'doesn''t have a key attribute.'
+            Write-Error -Message $msg -ErrorAction $ErrorActionPreference
+            return
+        }
+
+        $firstLine = "IIS configuration collection ${displayPath}"
+        $firstLineWritten = $false
+
+        $itemsRemoved = $false
+    }
+
+    process
+    {
+        if ($stopProcessing)
+        {
+            return
+        }
+
+        if (-not $firstLineWritten)
+        {
+            Write-Information $firstLine
+            $firstLineWritten = $true
+        }
+
+        foreach ($valueItem in $Value)
+        {
+            $itemToRemove = $collection | Where-Object { $_.GetAttributeValue($keyAttrName) -eq $valueItem }
+
+            if (-not $itemToRemove)
+            {
+                $msg = "Failed to remove item ""${valueItem}"" from IIS configuration collection ${displayPath} " +
+                       'because it doesn''t exist in the collection.'
+                Write-Error $msg -ErrorAction $ErrorActionPreference
+                return
+            }
+
+            Write-Information "  - $($valueItem)"
+            $collection.Remove($itemToRemove)
+            $itemsRemoved = $true
+        }
+    }
+
+    end
+    {
+        if ($stopProcessing -or -not $itemsRemoved)
+        {
+            return
+        }
+
+        if ($SkipCommit)
+        {
+            return $true
+        }
+
+        Save-CIisConfiguration

```

Comment:
Should `Save-CIisConfiguration` be updated to check `$SkipCommit` itself? That way the other functions can simply always call `Save-CIisConfiguration` and not worry about whether to actually commit the changes or not?

Reply:
This might be a future enhancement, but right now, only `Remove-CIisCollectionItem` and `Set-CIisCollectionItem` implement it. I'd have to write `Set-CIisSkipCommit` and `Reset-CIisSkipCommit` functions to manage it and then update `Set-CIisCollection` to call them both. Easier for now just to implement it as switches.